deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/scope/window/WindowContextImpl.java

```diff
@@ -82,6 +82,11 @@ public void activateWindow(String windowId)
     @Override
     public String getCurrentWindowId()
     {
+        if (windowHolder == null)
```

Comment:
it sounds like a broken setup, because a proxy to the windowHolder gets used. the init code (including the lookup of the proxy) is part of the startup process - see: https://github.com/apache/deltaspike/blob/master/deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/scope/DeltaSpikeContextExtension.java#L91

that means ds isn't in a valid state in this case and you might face further issues.